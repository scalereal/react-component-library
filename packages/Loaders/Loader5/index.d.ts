import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
export declare const SizeMap: Props['size'][];
declare const Loader5: React.FC<Props> & WithStyle;
export default Loader5;
