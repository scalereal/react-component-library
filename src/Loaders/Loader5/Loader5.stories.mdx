import Loader5 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader5"
    component={Loader5}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader5 {...args} />
    </div>
);

<Preview>
    <Story name="Loader5" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Loader5 size="S" Color="green" />
        <Loader5 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <Loader5 size="XS" />
        <Loader5 size="S" />
        <Loader5 size="M" />
        <Loader5 size="L" />
        <Loader5 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader5" />
