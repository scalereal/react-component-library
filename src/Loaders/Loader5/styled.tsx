import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    border: 0 solid transparent;
	border-radius: 50%;
	position: relative;
	&:before,
	&:after {
		content: '';
		border: .2em solid currentcolor;
		border-radius: 50%;
		width: inherit;
		height: inherit;
		position: absolute;
		top: 0;
		left: 0;
		animation: loader-07 1s linear infinite;
		opacity: 0;
	}
	&:before {
		animation-delay: 1s;
	}
	&:after {
		animation-delay: .5s;
	}

    @keyframes loader-07 {
        0% {
            transform: scale(0);
            opacity: 0;
        }
        50% {
            opacity: 1;
        }
        100% {
            transform: scale(1);
            opacity: 0;
        }
    }

`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 0 5px 5px 0;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 0 8px 8px 0;
            }
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 0 15px 15px 0;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 0 20px 20px 0;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 0 30px 30px 0;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 0 45px 45px 0;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin: 0 8px 8px 0;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.display?'inline-block':'none'};
`;
