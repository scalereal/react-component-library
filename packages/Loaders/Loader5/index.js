var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Loader, LoaderWrapper } from './styled';
export var SizeMap = ['XXS', 'XS', 'S', 'M', 'L', 'XL'];
var Loader5 = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(Loader, __assign({}, props))));
}));
Loader5.displayName = "Loader5";
Loader5.defaultProps = {
    size: 'S',
    display: true
};
export default Loader5;
