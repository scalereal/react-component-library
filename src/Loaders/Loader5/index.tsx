import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

export const SizeMap: Props['size'][] = ['XXS','XS', 'S', 'M', 'L', 'XL'];

const Loader5:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

 Loader5.displayName="Loader5";
 Loader5.defaultProps = {
    size:'S',
    visible: true
};

export default Loader5;
