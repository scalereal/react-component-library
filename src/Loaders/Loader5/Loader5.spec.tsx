import { render } from '@testing-library/react';
import * as React from 'react';
import Loader5 from '.';

describe('Loader5 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(< Loader5/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            < Loader5
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            < Loader5 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});