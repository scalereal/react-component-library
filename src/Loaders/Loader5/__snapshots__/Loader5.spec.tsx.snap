// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader5 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz kNzTru"
  >
    <div
      class="sc-bdfBwQ gmwuAc"
    />
  </div>
</div>
`;

exports[`Loader5 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz cvISIw"
  >
    <div
      class="sc-bdfBwQ hNEEYx"
    />
  </div>
</div>
`;

exports[`Loader5 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz vwiht"
  >
    <div
      class="sc-bdfBwQ gmwuAc"
    />
  </div>
</div>
`;
